Zend/zend_object_handlers.c

```diff
@@ -275,30 +275,6 @@ static ZEND_COLD zend_never_inline void zend_bad_property_name(void) /* {{{ */
 }
 /* }}} */
 
-static ZEND_COLD zend_never_inline void zend_forbidden_dynamic_property(
-		zend_class_entry *ce, zend_string *member) {
-	zend_throw_error(NULL, "Cannot create dynamic property %s::$%s",
-		ZSTR_VAL(ce->name), ZSTR_VAL(member));
-}
-
-static ZEND_COLD zend_never_inline bool zend_deprecated_dynamic_property(
-		zend_object *obj, zend_string *member) {
-	GC_ADDREF(obj);
-	zend_error(E_DEPRECATED, "Creation of dynamic property %s::$%s is deprecated",
-		ZSTR_VAL(obj->ce->name), ZSTR_VAL(member));
-	if (UNEXPECTED(GC_DELREF(obj) == 0)) {
-		zend_class_entry *ce = obj->ce;
-		zend_objects_store_del(obj);
-		if (!EG(exception)) {
-			/* We cannot continue execution and have to throw an exception */
-			zend_throw_error(NULL, "Cannot create dynamic property %s::$%s",
```

Comment:
The use cases for opcode handlers deprecating dynamic properties and for unserialize deprecating dynamic properties are different and shouldn't be unified

1. As mentioned in my other comment, the reference counting edge case handling only matters for the opcode handlers
2. For the unserialize() handler, this should probably fail on **any** `EG(exception)` after `zend_error(E_DEPRECATED` - see https://github.com/igbinary/igbinary/pull/362/files#diff-57bd563fa14d413e9e63720be6c9bc10ee7a72fe87b2936938f6d8be467f2c0e
3. zend_get_unmangled_prop_name makes sense for unserialize but not opcode handlers